### PR TITLE
Cache tokens and reuse them before expiry

### PR DESCRIPTION
**Pull request by ffilippopoulos**
https://developers.google.com/drive/api/v3/quickstart/go

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -124,6 +124,14 @@ func getAgentConfigPathFromHome() string {
 	return path.Join(home, "wiresteward.json")
 }
 
+func getAgentTokFilePathFromHome() string {
```
Comment (ribbybibby) [inline main.go:127]:
```suggestion
func getAgentTokenFilePathFromHome() string {
```

Unless there's some convention here I'm unaware of, I think using `Token` is clearer.

Comment (george-angel) [inline main.go:127]: We've had issues with HOME before, might want to have a look at: https://github.com/uw-labs/strongbox/blob/master/strongbox.go#L115

File: main.go [hunk 2]
```diff
@@ -146,6 +154,7 @@ func agent() {
 		agentConf.Oidc.AuthUrl,
 		agentConf.Oidc.TokenUrl,
 		agentConf.Oidc.ClientID,
+		getAgentTokFilePathFromHome(),
```
Comment (ribbybibby) [inline main.go:157]:
```suggestion
		getAgentTokenFilePathFromHome(),
```

File: main.go (hunk 1 above)
Comment (george-angel) [reply main.go:127]: Also "get" is very anti-pattern in Go

Comment (alkar) [reply main.go]: I've had troubles with that in the past in darwin, I've successfully used https://github.com/mitchellh/go-homedir

**Comment by alkar** [comment]
If we're going to have multiple files in our home directory, should we organise based on XDG user directories?
Namely, `os.UserConfigDir` for config and `os.UserCacheDir` for token?

**Comment by george-angel** [comment]
> If we're going to have multiple files in our home directory, should we organise based on XDG user directories?
> Namely, `os.UserConfigDir` for config and `os.UserCacheDir` for token?

Oh please no, thats all sorts can of worms.

I agree in principle, makes me uneasy having two files knocking around home dir, but perhaps not XDG as solution.

**Pull request merged**